Adição das funcionalidades de parcela e lembrete

Yasmiinmuniz opened the pull request:
controller, service, repository e fachada

Yasmiinmuniz committed:
Adição das funcionalidades de parcela e lembrete: controller, service, repository e fachada

Yasmiinmuniz committed:
Adição de transacao

jacksonlmp commented on src/main/java/br/edu/ufape/agiota/controllers/LembreteController.java:
```diff
@@ -0,0 +1,46 @@
+        try {
+            Lembrete lembrete = fachada.buscarLembrete(id);
+            return ResponseEntity.ok().body(lembrete);
+        } catch (RegistroNaoEncontradoException e) {
```
Lançar outra exceção por exemplo, "RegistroJaExistenteException".
Não é o caso de não encontrar registro

ThiagoCavalcantiSilva replied:
A observação seria na rota de criar lembrete, porém, ainda não foi resolvido

jacksonlmp commented on src/main/java/br/edu/ufape/agiota/controllers/TransacaoController.java:
```diff
@@ -0,0 +1,46 @@
+        try {
+            Transacao transacao = fachada.criarTransacao(transacaoDTO);
+            return ResponseEntity.ok().body(transacao);
+        } catch (RegistroNaoEncontradoException e) {
```
Lançar outra exceção por exemplo, "RegistroJaExistenteException".
Não é o caso de não encontrar registro

ThiagoCavalcantiSilva replied:
Ainda não resolvido

jacksonlmp commented on src/main/java/br/edu/ufape/agiota/dtos/TransacaoDTO.java:
```diff
@@ -0,0 +1,71 @@
+    @NotNull
+    private Date dataTransacao;
+
+	private Date data;
```
Corrigir espaçamento

jacksonlmp commented on src/main/java/br/edu/ufape/agiota/dtos/TransacaoDTO.java:
```diff
@@ -0,0 +1,71 @@
+
+	private Date data;
+
+	private Parcela parcela;
```
Corrigir espaçamento

jacksonlmp commented on src/main/java/br/edu/ufape/agiota/fachada/Fachada.java:
```diff
@@ -44,26 +56,48 @@ public Cliente atualizarCliente(ClienteDTO clienteDTO, long id) throws RegistroN
+        return transacaoService.buscarTransacoesPorParcela(idParcela);
     }
 }
+}
```
Colchete a mais, impedindo o projeto de rodar

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/controllers/LembreteController.java:
```diff
@@ -0,0 +1,46 @@
+    private Fachada fachada;
+
+    @GetMapping
+    public List<Lembrete> listarLembrete() {
```
Deveria buscar lembretes apenas de um usuário.
Verificar feature de avaliar reputação e replicar o código.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/controllers/ParcelaController.java:
```diff
@@ -0,0 +1,35 @@
+    private Fachada fachada;
+
+    @GetMapping
+    public List<Parcela> listarParcelas() {
```
Deveria buscar lembretes apenas de um empréstimo.
Verificar feature de avaliar reputação e replicar o código.

ThiagoCavalcantiSilva commented on git:
Remover arquivo vazio

ThiagoCavalcantiSilva commented on FETCH_HEAD:
Remover arquivo vazio

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/controllers/ParcelaController.java:
```diff
@@ -0,0 +1,35 @@
+    @GetMapping("/{id}")
+    public ResponseEntity<?> buscarParcela(@PathVariable long id) {
+        try {
+            Parcela parcela = (Parcela) fachada.buscarTransacoesPorParcela(id);
```
A conversão para parcela está incorreta. O método retorna uma lista de transações, não há como ser convertido em parcela.
Ajustar o método chamado e criar um endpoint em TransacaoController para buscar as transações da parcela.

ThiagoCavalcantiSilva replied:
Faltou criar a rota em TransacaoController chamando o método "buscarTransacoesPorParcela", que ficou sem uso na fachada

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/dtos/LembreteDTO.java:
```diff
@@ -0,0 +1,24 @@
+    private String texto;
+    private long parcelaId;
+
+    public Date getData() {
+        return data;
+    }
+
+    public String getTexto() {
+        return texto;
+    }
+
+    public long getParcelaId() {
+        return parcelaId;
+    }
```
Remover getters (já estão contidos na anotação @Data do lombok).
Adicionar anotações de validação no DTO e o @Valid no controller que o utiliza.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/dtos/TransacaoDTO.java:
```diff
@@ -0,0 +1,71 @@
+        return data;
+    }
+
+    @NotNull
+    public void setData(Date data) {
+        this.data = data;
+    }
+
+    @NotNull
+    public BigDecimal getValor() {
+        return valor;
+    }
+
+    public void setValor(BigDecimal valor) {
+        this.valor = valor;
+    }
+
+    public String getMetodoPagamento() {
+        return metodoPagamento;
+    }
+
+    public void setMetodoPagamento(String metodoPagamento) {
+        this.metodoPagamento = metodoPagamento;
+    }
+
+    public Date getDataTransacao() {
+        return dataTransacao;
+    }
+
+    public void setDataTransacao(Date dataTransacao) {
+        this.dataTransacao = dataTransacao;
+    }
+
+    public Parcela getParcela() {
+        return getParcela();
+    }
+
+    public void setParcela(Parcela parcela) {
+        this.parcela = parcela;
+    }
```
Remover getters e setters, pois a classe já está anotada com @Data.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/fachada/Fachada.java:
Retornar o código removido de outras features (services e métodos de agiota e avaliação).

ThiagoCavalcantiSilva replied:
Ainda não resolvido

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/basica/Lembrete.java:
```diff
@@ -16,4 +17,8 @@ public class Lembrete {
     
     @ManyToOne
     private Parcela parcela;
+
+	public static List<Lembrete> listarLembrete() {
+		return null;
+	}
```
Remover método e import da lista.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/interfaces/TransacaoServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+
+    Transacao buscarTransacao(long idTransacao) throws RegistroNaoEncontradoException;
+
+	List<Transacao> listarTransacao();
```
Não será necessário listar todas as transações de todos os usuários/empréstimos/parcelas. Deverá listar transações de um determinado empréstimo, por exemplo.

ThiagoCavalcantiSilva replied:
Ainda não resolvido

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/interfaces/TransacaoServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+
+public interface TransacaoServiceInterface {
+    Transacao criarTransacao(TransacaoDTO transacaoDTO) throws RegistroNaoEncontradoException;
```
Substituir exceção pela "RegistroJaExistenteException"

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/interfaces/ParcelaServiceInterface.java:
```diff
@@ -0,0 +1,14 @@
+import br.edu.ufape.agiota.negocio.basica.Parcela;
+
+public interface ParcelaServiceInterface {
+    List<Parcela> Parcela = null;
+	static List<Parcela> listarParcelas() {
+		return Parcela;
+	}
```
Remover método, pois não segue o padrão e apenas devolve uma lista nula.

ThiagoCavalcantiSilva replied:
Ainda não resolvido

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/interfaces/LembreteServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+
+public interface LembreteServiceInterface {
+    List<Lembrete> listarLembrete();
```
Deve filtrar pelo usuarioId.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/interfaces/LembreteServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+
+    Lembrete buscarLembrete(long id) throws RegistroNaoEncontradoException;
+
+    Lembrete atualizarLembrete(LembreteDTO lembreteDTO, long id) throws RegistroNaoEncontradoException;
```
Não será necessário atualizar lembretes.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/interfaces/LembreteServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+public interface LembreteServiceInterface {
+    List<Lembrete> listarLembrete();
+
+    Lembrete criarLembrete(LembreteDTO lembreteDTO) throws RegistroNaoEncontradoException;
```
Utilizar a "RegistroJaExistenteException"

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/basica/Transacao.java:
```diff
@@ -19,5 +20,52 @@ public class Transacao {
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public Date getData() {
+        return data;
+    }
+
+    public void setData(Date data) {
+        this.data = data;
+    }
+
+    public BigDecimal getValor() {
+        return valor;
+    }
+
+    public void setValor(BigDecimal valor) {
+        this.valor = valor;
+    }
+
+    public String getMetodoPagamento() {
+        return metodoPagamento;
+    }
+
+    public void setMetodoPagamento(String metodoPagamento) {
+        this.metodoPagamento = metodoPagamento;
+    }
+
+    public Date getDataTransacao() {
+        return dataTransacao;
+    }
+
+    public void setDataTransacao(Date dataTransacao) {
+        this.dataTransacao = dataTransacao;		
+	}
+
```
Remover getters e setters

Yasmiinmuniz replied:
certeza?

ThiagoCavalcantiSilva replied:
SIm, já estão inclusos na anotação "@Data" do lombok.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/basica/Transacao.java:
```diff
@@ -19,5 +20,52 @@ public class Transacao {
+    @JoinColumn(name = "parcela_id")
+    private Parcela parcela;
+    
+    public void setParcela(Parcela parcela) {
```
Remover método set

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/basica/Transacao.java:
```diff
@@ -19,5 +20,52 @@ public class Transacao {
+        this.dataTransacao = dataTransacao;		
+	}
+
+    @JoinColumn(name = "parcela_id")
```
Substituir o joinColumn pela anotação de relacionamento com cardinalidade (@OneToOne, @OneToMany, @ManyToOne ou @ManyToMany).

ThiagoCavalcantiSilva replied:
Ainda não resolvido

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/ParcelaService.java:
```diff
@@ -0,0 +1,35 @@
+    @Autowired
+    private ParcelaRepository parcelaRepository;
+
+    public List<Parcela> listarParcelas() {
+        return parcelaRepository.findAll();
```
Filtrar pelo empréstimo

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/LembreteService.java:
```diff
@@ -0,0 +1,51 @@
+    private LembreteRepository lembreteRepository;
+
+    @Override
+    public List<Lembrete> listarLembrete() {
+        return lembreteRepository.findAll();
```
Filtrar pelo usuário

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/LembreteService.java:
```diff
@@ -0,0 +1,51 @@
+
+        if (parceOpt.isPresent()) return parceOpt.get();
+
+        throw new RegistroNaoEncontradoException("Lembrete com o identificador " + id + " não foi encontrada!");
```
```suggestion
        throw new RegistroNaoEncontradoException("Lembrete com o identificador " + id + " não foi encontrado!");
```

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/LembreteService.java:
```diff
@@ -0,0 +1,51 @@
+
+        throw new RegistroNaoEncontradoException("Lembrete com o identificador " + id + " não foi encontrada!");
+    }
+    public Lembrete atualizarCliente(LembreteDTO lembreteDTO, long id) throws RegistroNaoEncontradoException {
+        Lembrete lembrete = buscarLembrete(id);
+        return lembreteRepository.save(lembrete);
```
O nome do método não se refere ao contexto da classe.
E o corpo do método atualiza o objeto que acabou de consultar do banco, sem fazer alterações nele.

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/LembreteService.java:
```diff
@@ -0,0 +1,51 @@
+        return lembreteRepository.save(lembrete);
+    }
+
+	@Override
+	public Lembrete criarLembrete(LembreteDTO LembreteDTo) throws RegistroNaoEncontradoException {
+		return null;
+	}
+
+	@Override
+	public Lembrete atualizarLembrete(LembreteDTO c, long id) throws RegistroNaoEncontradoException {
+		return null;
+	}
```
Remover métodos, pois não foram implementados

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/controllers/TransacaoController.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/transacao")
```
Colocar nome no plural (e ajustar na coleção do insomnia)

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/controllers/TransacaoController.java:
```diff
@@ -0,0 +1,46 @@
+    private Fachada fachada;
+
+    @GetMapping
+    public List<Transacao> listarTransacao() {
+        return fachada.listarTransacoes();
+    }
```
Filtrar pelo empréstimo ou parcela, a depender do que fizer mais sentido

ThiagoCavalcantiSilva replied:
Ainda não resolvido

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/TransacaoService.java:
```diff
@@ -0,0 +1,64 @@
+    @Autowired
+    private ParcelaRepository parcelaRepository;
+
+    @Override
+    public List<Transacao> listarTransacao() {
+        return transacaoRepository.findAll();
+    }
```
Não deve listar todas as transações existentes, deve filtrar por algo (usuário ou empréstimo, por exemplo). A depender de como o sistema será arquitetado

ThiagoCavalcantiSilva replied:
Ainda não resolvido

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/TransacaoService.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    @Override
+    public Transacao criarTransacao(TransacaoDTO transacaoDTO) throws RegistroNaoEncontradoException {
```
Substituir por "RegistroJaExistenteException"

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/negocio/services/TransacaoService.java:
```diff
@@ -0,0 +1,64 @@
+        }
+        Parcela parcela = parcelaOpt.get();
+
+        Transacao transacao = new Transacao();
+        transacao.setData(new Date());
+        transacao.setValor(transacaoDTO.getValor());
+        transacao.setMetodoPagamento(transacaoDTO.getMetodoPagamento());
+        transacao.setDataTransacao(transacaoDTO.getDataTransacao());
+        transacao.setParcela(parcela);
+
```
Mover alteraçõs dos atributos para dentro do DTO

ThiagoCavalcantiSilva commented on src/main/java/br/edu/ufape/agiota/dtos/ParcelaDTO.java:
Remover arquivo não utilizado

jacksonlmp committed:
Correção fachada

Yasmiinmuniz committed:
"Renomear endpoint de /transacao para /transacoes e ajustar a coleção no Insomnia"

Yasmiinmuniz committed:
"Remover ParcelaDTO.java"

Yasmiinmuniz committed:
"Atualizar Transacao para adicionar método criarTransacao e remover getters/setters duplicados, alterar a listagem para procurar por emprestimo ou parcela"

Yasmiinmuniz committed:
Implementar método criarLembrete e atualizarLembrete em LembreteService e adicionar conversão em LembreteDTO

Yasmiinmuniz committed:
Corrigir métodos em LembreteService e Fachada, adicionar métodos necessários em LembreteRepository

Yasmiinmuniz committed:
"Renomear ListarLembrete para ListaLembretes"

Yasmiinmuniz committed:
Refatoração da classe Fachada para corrigir métodos de listagem estática e instância de serviços

- Corrigido método listarParcelas() para usar a instância parcelasService ao invés da interface estática ParcelaServiceInterface.listarParcelas().
- Atualizado método listarLembretes() para corresponder à assinatura correta em LembreteServiceInterface.
- Adicionado getter e setter para parcelasService na classe Fachada para garantir a injeção correta de dependência.
- Ajustadas outras dependências autowired para os serviços de cliente, agiota, avaliação, lembrete e transação.

Yasmiinmuniz committed:
"Altereções no nome dos métodos"
- listarLembrete | listarLembretes
- listarTransacao | listarTransacoes
- parcela | parcelas

Yasmiinmuniz committed:
"Refatoração do ParcelaService para adicionar método listarParcelasPorEmprestimo que busca parcelas por empréstimo usando ParcelaRepository."

Yasmiinmuniz committed:
Corrigido o relacionamento de um-para-um na classe `Transacao`, substituindo `@OnetoOne` por `@OneToOne` e ajustando a importação correta para `jakarta.persistence`.

Yasmiinmuniz committed:
Corrigido erro de sintaxe na classe Transacao: adicionado ponto e vírgula após declaração do campo parcela no mapeamento com OneToOne.

Yasmiinmuniz committed:
Remoção dos getters e setters transacao.java

Yasmiinmuniz committed:
Atualização no serviço de Lembretes:
- Adicionado método listarLembretesPorUsuarioId no LembreteServiceInterface.
- Implementado método listarLembretesPorUsuarioId no LembreteService.
- Adicionado método findByUsuarioId no LembreteRepository.
- Atualizada a Fachada para utilizar o novo método listarLembretesPorUsuarioId.
- Adicionado suporte para listar lembretes por usuarioId no método listarLembretesPorUsuarioId.

Yasmiinmuniz committed:
Alterações e correções no método listarParcelasPorEmprestimo

Yasmiinmuniz committed:
Substituir exceção de criarTransacao pela RegistroJaExistenteException

Yasmiinmuniz committed:
Remoção de arquivos vázios

Yasmiinmuniz committed:
Merge branch 'main' of https://github.com/jacksonlmp/agiota_web into feature/lembrete_parcela_transacao

Yasmiinmuniz committed:
ajuste na linha

Yasmiinmuniz committed:
Merge branch 'feature/lembrete_parcela_transacao' of https://github.com/jacksonlmp/agiota_web into feature/lembrete_parcela_transacao

Yasmiinmuniz committed:
Troca na exceção de criarLembrete

Yasmiinmuniz committed:
Organização, ajustes e adição de funções do emprestimo na fachada

Yasmiinmuniz committed:
Rota em TransacaoController chamando o método buscarTransacoesPorParcela

Yasmiinmuniz committed:
Rota em Transacao para o método listarTransacoesPorEmprestimo

Yasmiinmuniz committed:
Troca de excessão em Transação"

Yasmiinmuniz committed:
Remoção do @joinColumn de Transação

Yasmiinmuniz committed:
Remoção do método ListarParcelas(), pois retornava vázio
